dev/com.ibm.ws.threading_policy_fat/test-applications/basicfat/src/web/PolicyExecutorServlet.java

```diff
@@ -2397,6 +2397,183 @@ public void testInvokeAllTimeoutWaitForEnqueue() throws Exception {
+        PolicyExecutor executor = provider.create("testInvokeAnyTimedAllTimeout");
+
+        final int numTasks = 3;
+
+        CountDownLatch unused = new CountDownLatch(0);
+        CountDownLatch blocker = new CountDownLatch(1);
+        List<Callable<Boolean>> tasks = new LinkedList<Callable<Boolean>>();
+        for (int i = 0; i < numTasks; i++)
+            tasks.add(new CountDownTask(unused, blocker, TIMEOUT_NS * 2));
+
+        long start = System.nanoTime();
+        try {
+            boolean result = executor.invokeAny(tasks, 200, TimeUnit.MILLISECONDS);
+            fail("invokeAny should time out, instead: " + result);
+        } catch (TimeoutException x) { // pass
+        }
+        long duration = System.nanoTime() - start;
+
+        assertTrue(duration + "ns", duration < TIMEOUT_NS); // higher value used to allow for slowness and pauses in test systems
+
+        List<Runnable> canceledFromQueue = executor.shutdownNow();
+        assertEquals(0, canceledFromQueue.size());
+    }
+
+    // Submit a group of tasks to timed invokeAny, where there are insufficient queue positions to
+    // enqueue the tasks and the timeout occurs while waiting or a queue position to become available
+    // rather than while running the task.
+    @Test
+    public void testInvokeAnyTimedTimeoutDuringWaitForEnqueue() throws Exception {
+        PolicyExecutor executor = provider.create("testInvokeAnyTimedTimeoutDuringWaitForEnqueue")
+                        .maxConcurrency(1)
+                        .maxQueueSize(1)
+                        .maxWaitForEnqueue(TimeUnit.MINUTES.toMillis(4))
+                        .queueFullAction(QueueFullAction.Abort);
+
+        AtomicInteger counter = new AtomicInteger();
+
+        // Submit blocking task to use up the single maxConcurrency permit
+        CountDownLatch unused = new CountDownLatch(0);
+        CountDownLatch blocker = new CountDownLatch(1);
+        Future<Boolean> future1 = executor.submit(new CountDownTask(unused, blocker, TimeUnit.MINUTES.toNanos(5)));
+
+        // Submit another task to use up the single queue position
+        Future<?> future2 = executor.submit((Runnable) new SharedIncrementTask(counter));
+
+        final int numInvokeAnyTasks = 2;
+        List<Callable<Integer>> tasks = new LinkedList<Callable<Integer>>();
+        for (int i = 0; i < numInvokeAnyTasks; i++)
+            tasks.add(new SharedIncrementTask(counter));
+
+        long start = System.nanoTime();
+        try {
+            int result = executor.invokeAny(tasks, 200, TimeUnit.MILLISECONDS);
+            fail("invokeAny should be rejected when unable to submit task, instead: " + result);
+        } catch (RejectedExecutionException x) { // pass
+        }
+        long duration = System.nanoTime() - start;
+
+        assertTrue(duration + "ns", duration < TIMEOUT_NS); // higher value used to allow for slowness and pauses in test systems
```

Comment:
If the timeout is 200 ms, should we be checking the duration for something less than TIMEOUT_NS, which is 2 minutes?

Reply:
See earlier comment.